src/Features/LanguageServer/Protocol/RoslynRequestExecutionQueue.cs

```diff
@@ -39,6 +44,48 @@ public override Task WrapStartRequestTaskAsync(Task nonMutatingRequestTask, bool
             }
         }
 
+        /// <inheritdoc/>
+        protected override IMethodHandler GetHandlerForRequest(IQueueItem<RequestContext> work)
+        {
+            var handlers = _handlerProvider.GetMethodHandlers(work.MethodName, work.RequestType, work.ResponseType);
+            var defaultHandler = handlers.First(h => string.IsNullOrEmpty(h.Metadata)).Value;
+            var identifier = RoslynRequestExecutionQueue.GetTextDocumentIdentifier(work, defaultHandler);
+            if (identifier is null)
+            {
+                return defaultHandler;
+            }
+
+            var language = _lspWorkspaceManager.GetLanguageForUri(identifier.Uri);
+
+            return handlers.SingleOrDefault(h => string.Equals(h.Metadata, language, StringComparison.OrdinalIgnoreCase))?.Value ?? defaultHandler;
```

Comment:
why ordinalignore case?

Reply:
also, i realy don't like all this code that just fallsback when things fail.

Reply:
LSP languages are supposed to be lowercase. However, currently XAML is reported by VS Code as "XAML" from the MAUI extension, so I found it safer to ignore the casing.

Reply:
i would much rather we not add ignore casing into the system.  it tends to just lead to bugs in my experience.  we haven't had to so anything like that so far for the c# name we use everwwhere.

Reply:
+1 - I mentioned above, I think we can be explicit about the language for all languages without too many changes (if we update the MethodAttribute to default to C#/VB/F# and rename in a separate PR)

Reply:
This is now handled via ILanguageInfoProvider